pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -57,13 +59,16 @@
  * <p>There could be multiple DIMENSION or METRIC or DATE_TIME fields, but at most 1 TIME field.
  * <p>In pinot, we store data using 5 <code>DataType</code>s: INT, LONG, FLOAT, DOUBLE, STRING. All other
  * <code>DataType</code>s will be converted to one of them.
+ *
+ * <p>The schema can be configured with different {@link Options}, including how to handle nulls</p>
  */
 @SuppressWarnings("unused")
 @JsonIgnoreProperties(ignoreUnknown = true)
 public final class Schema implements Serializable {
   private static final Logger LOGGER = LoggerFactory.getLogger(Schema.class);
 
   private String _schemaName;
+  private Options _options = new Options();
```

Comment:
Can we keep it simple by just adding a `boolean enableColumnBasedNullHandling` and default to false? This should be a short lived flag for migration, so I'd suggest keeping it simple

Reply:
I like the proposed approach, but I've simplified to what Jackie proposed in 11e597f4fe776c53ebaafbef3bed3c68446c0b96